### ISSUE TITLE
DCGM initialization error

### ISSUE THREAD

**Issue by Rusteam** [triton-inference-server/model_analyzer#583]
I'm trying to run model-analyzer with `tritonserver:22.12-py-sdk` (also `22.09`) container on kubernetes cluster with a Tesla T4 GPU. However I get the following error:

```
Error: Failed to initialize NVML
Traceback (most recent call last):
  File "/usr/local/bin/model-analyzer", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/model_analyzer/entrypoint.py", line 251, in main
    gpus = GPUDeviceFactory().verify_requested_gpus(config.gpus)
  File "/usr/local/lib/python3.8/dist-packages/model_analyzer/device/gpu_device_factory.py", line 36, in __init__
    self.init_all_devices()
  File "/usr/local/lib/python3.8/dist-packages/model_analyzer/device/gpu_device_factory.py", line 55, in init_all_devices
    dcgm_handle = dcgm_agent.dcgmStartEmbedded(
  File "/usr/local/lib/python3.8/dist-packages/model_analyzer/monitor/dcgm/dcgm_agent.py", line 41, in dcgmStartEmbedded
    dcgm_structs._dcgmCheckReturn(ret)
  File "/usr/local/lib/python3.8/dist-packages/model_analyzer/monitor/dcgm/dcgm_structs.py", line 646, in _dcgmCheckReturn
    raise DCGMError(ret)
model_analyzer.monitor.dcgm.dcgm_structs.DCGMError_InitError: DCGM initialization error
```
as it's unable to detect a GPU:
```
WARNING: The NVIDIA Driver was not detected.  GPU functionality will not be available.
   Use the NVIDIA Container Toolkit to start this container with GPU support; see
   https://docs.nvidia.com/datacenter/cloud-native/ .
```

Although the base server `tritonserver:22.09-py` runs just fine with kubernetes on the same GPU. I follow the guide outlined [here](https://github.com/triton-inference-server/model_analyzer/blob/main/docs/kubernetes_deploy.md)

**Comment by mc-nv** [comment]
Hi @Rusteam  
Could you please share K8S cluster configuration details. In particular I'm interested about the output for the command:
`kubectl get pods -A`.

**Comment by Rusteam** [comment]
I've solved this issue by adding `runtime: nvidia` to the job template. I can make a PR

**Comment by mc-nv** [comment]
Hi @Rusteam 

Aftere reviewing the documentation guidance [Using Kubernetes with GPUs](https://github.com/triton-inference-server/model_analyzer/blob/main/docs/kubernetes_deploy.md#using-kubernetes-with-gpus) for Model Analyzer. I don't see a reason to bring mentioned change to the Kubernetes template. <br>
I see Model Analyzer instructions explicetely mention set the default container runtime to `nvidia` to the Node which should not give any issue in case if Kubernetes Node has been cofigured with default runtime `nvidia`. <br>

Please feel free that approach [Assigning Pods to Nodes](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/) which is best fits to your Kubernetes cluster.

**Comment by Rusteam** [comment]
Maybe our node is not configured to use nvidia gpus by default. I just want to offer a solution to other people who might run into this issue as well. If not necessary, it's fine by me.
